Go defaults to "0"" so in case we want to return

jgheewala opened the pull request:
EntryStatus back to the caller "Expired" cannot be differentiated.
Fixing this by default "_" to 0 and incremental RemoveReasons

comment was missing for GetWithInfo api so updated that

test: ran all unit test cases

jgheewala committed:
Go defaults to "0"" so in case we want to return
EntryStatus back to the caller "Expired" cannot be differentiated.
Fixing this by default "_" to 0 and incremental RemoveReasons

comment was missing for GetWithInfo api so updated that

test: ran all unit test cases

jgheewala committed:
Merge branch 'master' into expired_reason_iota_fix

mxplusb commented:
I quite like this PR and problem it solves, but I agree this is definitely a 3.0 decision since it could break a lot of downstream configurations when they upgrade. This should definitely go in the release notes.

jgheewala commented:
@mxplusb this PR can be open until its ready to go in. I kept it here so that I do not forget about this :)

mxplusb commented:
Great, thanks!

flisky commented:
So, the error `ErrEntryIsDead` has no more referenced and should be removed ? Will give users a compile error in case they don't notice this breaking change.

cristaloleg commented:
@flisky good point, we can remove it. But anyway, this PR is for a next major version, so it will be safe to make a breaking change.

mxplusb commented:
@jgheewala when you get a chance, can you rebase? It would be good to keep this updated to prep for the next major version.

flisky commented:
May I take over of it, @jgheewala & @mxplusb ?

Besides these changes, I also want to return the expired entry even when `resp.EntryStatus = Expired`( `entry` goes `[]byte` when introduced in #168 , and then goes to `nil` in #191 ). Do you like the idea?

cristaloleg commented:
@flisky let's create a separate issue for your idea and discuss it there. This will make discussion much more visible and structured, thanks :)

jgheewala commented:
@mxplusb was super busy will rebase my changes soon

jgheewala committed:
Merge branch 'master' into expired_reason_iota_fix

jgheewala commented:
@mxplusb done with the conflicts.

siennathesane committed:
Merge branch 'master' into expired_reason_iota_fix

jgheewala committed:
Merge branch 'master' into expired_reason_iota_fix

mxplusb commented:
@jgheewala there's a new branch, called `versions/v3-prep` which has the underlying necessary module changes to support this PR. If you can rebase against that branch, and then edit the PR to point to the `versions/v3-prep` branch, I can get this merged and we can close it out.

jgheewala committed:
Merge branch 'master' into expired_reason_iota_fix

jgheewala commented:
@mxplusb https://github.com/allegro/bigcache/pull/255

Commit:
Entry RemoveReason is iota based with default value being 0

Currently all const for RemovedReason are explicitly
set to avoid any breaking changes. For v3 migrating the reasons
to iota makes code readability and managing addition of more reasons
easier in the future.

Test: Validated all test cases run successfully.

janisz committed:
Merge branch 'main' into expired_reason_iota_fix
